lib/vrl/compiler/src/expression/array.rs

```diff
@@ -54,6 +55,22 @@ impl Expression for Array {
 
         TypeDef::new().array(type_defs).with_fallibility(fallible)
     }
+
+    fn compile_to_vm(&self, vm: &mut crate::vm::Vm) -> Result<(), String> {
+        // Evaluate each of the elements of the array, the result of each
+        // will be added to the stack.
+        for value in self.inner.iter().rev() {
+            value.compile_to_vm(vm)?;
+        }
+
+        vm.write_opcode(OpCode::CreateArray);
+
+        // Add the length of the array as a primitive so the VM knows how
+        // many elements to move into the array.
+        vm.write_primitive(self.inner.len());
```

Comment:
The order does matter.

When the `CreateArray` opcode is executed, the VM expects the stack to be populated with the elements that it will use to build up the array.